src/Standards/Generic/Sniffs/Arrays/DisallowCompactArrayBuilderSniff.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/**
+ * Bans the use of compact() function
+ *
+ * @author    Paweł Bogut <pbogut@pbogut.me>
+ * @copyright 2006-2023 Squiz Pty Ltd (ABN 77 084 670 600)
+ * @license   https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Standards\Generic\Sniffs\Arrays;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use PHP_CodeSniffer\Util\Tokens;
+
+class DisallowCompactArrayBuilderSniff implements Sniff
+{
+
+
+    /**
+     * Registers the tokens that this sniff wants to listen for.
+     *
+     * @return int[]
+     */
+    public function register()
+    {
+        return [T_STRING];
+
+    }//end register()
+
+
+    /**
+     * Processes this test, when one of its tokens is encountered.
+     *
+     * @param \PHP_CodeSniffer\Files\File $phpcsFile The file being scanned.
+     * @param int                         $stackPtr  The position of the current token
+     *                                               in the stack passed in $tokens.
+     *
+     * @return void
+     */
+    public function process(File $phpcsFile, $stackPtr)
+    {
+        $tokens = $phpcsFile->getTokens();
+
+        $content = $tokens[$stackPtr]['content'];
+
+        if ($content !== 'compact') {
+            return;
+        }
+
+        // Make sure this is a function call.
+        $next = $phpcsFile->findNext(Tokens::$emptyTokens, ($stackPtr + 1), null, true);
+        if ($next === false || $tokens[$next]['code'] !== T_OPEN_PARENTHESIS) {
+            // Not a function call.
+            return;
+        }
+
+        $prev = $phpcsFile->findPrevious(Tokens::$emptyTokens, ($stackPtr - 1), null, true);
+
+        // Make sure its not method of an object or class.
+        if ($tokens[$prev]['code'] === T_OBJECT_OPERATOR || $tokens[$prev]['code'] === T_DOUBLE_COLON) {
```

Comment:
Yes, thanks. I've added that, plus few other things. 
I'm now questioning if I should look back for things that indicates it's not actually builtin function call, or maybe should change approach and list things that indicates it is function call.

So what I mean by that is if its better to check (as it is now) :

```
 if ($tokens[$prev]['code'] === T_OBJECT_OPERATOR ... ) {
     return;
 }
 ````
 Or maybe would be better something like:
 ```
 if ($tokens[$prev]['code'] !== T_EQUAL ...) {
     return;
 }
 ```
 It might be more future proof if changed, but not sure if I wont end up with long list of allowed tokens.

Reply:
In my experience the current approach of bowing out of specific tokens is the correct one.
Changing the logic around would make the sniff much less stable as new tokens get added regularly (either via PHP itself or in PHPCS to account for changes in PHP) and this sniff would need to be updated each time that happens.